File: app/src/main/java/com/labessence/biotech/finaldilution/component/concentration/PercentageConcentration.java
package com.labessence.biotech.finaldilution.component.concentration;

/**
 * Project: FinalDilution
 * Created by dawid.chmielewski on 11/11/2017.
 */

public class PercentageConcentration extends Concentration {
    public PercentageConcentration(double amount) {
        super(amount, ConcentrationType.PERCENTAGE);
    }

    @Override
    public double calcDesiredMass(double volume, double molarMass) {
        return concentration * volume / 100;
    }

    @Override
    public double calcVolumeForDesiredMass(double mass, double molarMass) {
        return mass / concentration * 100;
    }
}

File: app/src/main/java/com/labessence/biotech/finaldilution/component/concentration/PercentageConcentration.kt
package com.labessence.biotech.finaldilution.component.concentration

/**
 * Project: FinalDilution
 * Created by dawid.chmielewski on 11/11/2017.
 */

class PercentageConcentration(amount: Double) : Concentration(amount, ConcentrationType.PERCENTAGE) {

    override fun calcDesiredMass(volume: Double, molarMass: Double?): Double {
        return concentration * volume / 100
    }

    override fun calcVolumeForDesiredMass(mass: Double, molarMass: Double?): Double {
        return mass / concentration * 100
    }
}
